### ISSUE TITLE
Phrasal searching matching substrings

### ISSUE THREAD

**Issue by joeytakeda** [projectEndings/staticSearch#71]
If you do a search in MoEML for `"remark"` you get results for remarkably, remarkable, remarks, et cetera: https://mapoflondon.uvic.ca/search.htm?q=%22remark%22. 

My immediate reaction is that this is a bug: if you put a term in quotation marks, then the search should only return contexts that match that term exactly.

**Comment by martindholmes** [comment]
IIRC we decided that if you quote a single word, what you're doing is creating a MUST CONTAIN, so "remark" becomes +remark. This was based on the idea that a phrasal search is a search for a phrase, i.e. more than one word. However, it's certainly the case that Google doesn't work this way, so perhaps we shouldn't either. So I agree we should change this.

**Comment by martindholmes** [comment]
Ah -- here's an interesting thing: search for "remarkable":

https://mapoflondon.uvic.ca/search.htm?q=%22remarkable%22

**Comment by joeytakeda** [comment]
Interesting, indeed! That suggests to me that we are treating all quoted things as phrases; I think that's confirmed by this bit of JS (though I could be wrong):

https://github.com/projectEndings/staticSearch/blob/9cb42a565cf2e4068a4f73d5582fa678380fabd4/js/ssSearch.js#L684-L706

So if I'm understanding this correctly, I think the solution is just to make the regex a bit more precise and matching on the word boundary. 

So this bit:

https://github.com/projectEndings/staticSearch/blob/9cb42a565cf2e4068a4f73d5582fa678380fabd4/js/ssSearch.js#L1463-L1464

Could become something like:

```
let rePhr = new RegExp("\\b" + self.terms[phr].str + "\\b", 'i'); 
```

**Comment by martindholmes** [comment]
I believe you're right. I'll test as soon as I get time, and also add an appropriate search to our test suite. Looks like what I did was neither one thing nor the other.

**Comment by martindholmes** [comment]
Worth remembering that this will actually cause problems for languages with no explicit word-boundaries.

**Comment by martindholmes** [comment]
Should be fixed in commit af2623d. Let's wait for a MoEML build to test it properly before closing.

**Comment by martindholmes** [comment]
MoEML testing shows it as working well. Closing.
